futurex_openedx_extensions/helpers/caching.py

```diff
@@ -71,3 +74,36 @@ def wrapped(*args: Any, **kwargs: Any) -> Dict[str, Any]:
 
         return wrapped
     return decorator
+
+
+def invalidate_cache(cache_name: str) -> None:
+    """
+    Invalidate a specific cache or all predefined caches.
+
+    - To invalidate a specific cache, provide the cache name.
+    - To reset all predefined caches, pass `"__all__"` as `cache_name`.
+
+    :param cache_name: The name of the cache to invalidate, or `"__all__"` to invalidate all predefined caches.
+    :type cache_name: str
+    :raises FXCodedException: If the provided `cache_name` is invalid (and not `"__all__"`).
+    """
+
+    def invalidate_single_cache(name: str) -> None:
+        """Invalidate the given cache."""
+        if cs.CACHE_NAMES.get(name) is None:
+            raise FXCodedException(
+                code=FXExceptionCodes.INVALID_INPUT,
+                message=f'Cache with {name} does not exist.'
+            )
+        cache.set(name, None)
```

Comment:
no need to raise an error
```suggestion
        cache.delete(name)
```